Sources/Amplitude/AMPStorage.m

```diff
@@ -119,18 +123,17 @@ + (NSMutableArray *)getEventsFromDisk:(NSString *)path {
 }
 
 + (NSDictionary *)JSONFromFile:(NSString *)path {
-    NSData *data;
-    if ([NSData dataWithContentsOfFile:path] != nil) {
-       data = [NSData dataWithContentsOfFile:path];
-    } else {
+    NSData *data = [NSData dataWithContentsOfFile:path];
+    if (data == nil) {
         NSString *emptyData = @"{ \"batch\": []}";
         data = [emptyData dataUsingEncoding:NSUTF8StringEncoding];
     }
 
-    NSError *err = NULL;
+    NSError *err = nil;
     NSDictionary *jsonAsDict = [NSJSONSerialization JSONObjectWithData:data options:kNilOptions error:&err];
-    if (err != NULL) {
+    if (err != nil || jsonAsDict == nil) {
```

Comment:
If anyone is wondering about this part, invalid json will error silently and then *jsonAsDict will be a nil pointer.

Reply:
Fail silently meaning that *err will also be nil

Reply:
Good catch! I was hoping error is not nil if it is invalid json